src/OrleansRuntime/GrainDirectory/LocalGrainDirectory.cs

```diff
@@ -86,12 +88,14 @@ internal class LocalGrainDirectory :
         internal readonly CounterStatistic UnregistrationsManyRemoteSent;
         internal readonly CounterStatistic UnregistrationsManyRemoteReceived;
 
-        
         public LocalGrainDirectory(Silo silo)
         {
             log = LogManager.GetLogger("Orleans.GrainDirectory.LocalGrainDirectory");
 
             MyAddress = silo.LocalMessageCenter.MyAddress;
+            isSiloNextInTheRing = (siloAddr, hash, excludeMySelf) => (siloAddr.GetConsistentHashCode() <= hash) &&
+                                                                     (!excludeMySelf || !siloAddr.Equals(MyAddress));
```

Comment:
Since this is not using any closure, what is the purpose of having a delegate field instead of a (more readable) proper method?